### ISSUE TITLE
Bug in fit method

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#501]
This fails:

``` python
model = BayesianModel([('A', 'B'), ('B', 'C'), ('D', 'C'), ('C', 'E')])
raw_data = np.random.randint(low=0, high=8, size=(1000, 5))
values = pd.DataFrame(raw_data, columns=['A', 'B', 'C', 'D', 'E'])
model.fit(values)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-16-bb20c048f671> in <module>()
      2 raw_data = np.random.randint(low=0, high=8, size=(1000, 5))
      3 values = pd.DataFrame(raw_data, columns=['A', 'B', 'C', 'D', 'E'])
----> 4 model.fit(values)

/home/ankur/jmlr/pgmpy/pgmpy/models/BayesianModel.py in fit(self, data, estimator_type)
    554         estimator = estimator_type(self, data)
    555 
--> 556         cpds_list = estimator.get_parameters()
    557         self.add_cpds(*cpds_list)
    558 

/home/ankur/jmlr/pgmpy/pgmpy/estimators/MLE.py in get_parameters(self)
     71                 state_counts = self.data.groupby([node] + self.model.predecessors(node)).count()
     72                 values = state_counts.iloc[:, 0].reshape(var_card,
---> 73                                                          np.product(parent_card))
     74                 cpd = TabularCPD(node, var_card, values,
     75                                  evidence=parents,

/home/ankur/anaconda3/envs/jmlr/lib/python3.4/site-packages/pandas/core/series.py in reshape(self, *args, **kwargs)
    815             return self
    816 
--> 817         return self._values.reshape(shape, **kwargs)
    818 
    819     def iget_value(self, i, axis=0):

ValueError: total size of new array must be unchanged
```

**Comment by sandeepkrjha** [comment]
Hey, I tried to analyze this bug when I faced an unexpected problem. I am using Python 3.4.3 on Anaconda 2.3.0 and when I did

``` python
>>> import pgmpy
>>> from pgmpy.models import BayesianModels
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'pgmpy.models'
```

I am a beginner and hence unable to understand what is the problem here with the installation.

**Comment by ankurankan** [comment]
@sandeepkrjha I have no idea why this is happening. How did you install pgmpy?

**Comment by kshitij10496** [comment]
@sandeepkrjha Try `conda install pgmpy`.

**Comment by khalibartan** [comment]
@kshitij10496  I don't think that would work because pgmpy is not present linux channels.

For installing pgmpy go to the directory where you have cloned pgmpy . In pgmpy directory run `python setup.py install`
Rest will be taken care of by script.
If somehow errors are given than manually install these packages in conda 

```
conda install python=3.4 numpy scipy pandas networkx pyparsing mock pip
```

after this install nose

```
pip install nose
```

and then try running setup.py again. This time It should definitely work.

**Comment by ankurankan** [comment]
Fixed in #541
